openwisp_controller/config/api/urls.py

```diff
@@ -0,0 +1,55 @@
+from django.conf import settings
+from django.urls import path
+
+from . import views as api_views
+
+app_name = 'openwisp_controller'
+
+
+def get_api_urls(api_views):
+    """
+    returns:: all the API urls of the config app
+    """
+    if getattr(settings, 'OPENWISP_CONTROLLER_API', True):
+        return [
+            path(
+                'controller/template/',
```

Comment:
| SN | /api/v1/controller/template/                               | Status |
|----|------------------------------------------------------------|--------|
| 1  | [SU] Can view all templates                                | Pass   |
| 2  | [O1A] can view only org1 templates (not even shared org)   | Pass   |
| 3  | [O1O] can view no templates                                | Pass   |
| 4  | [SU] VPN-Client hidden for Generic Template                | NTD   |
| 5  | [SU][O1A] Field validation error on field                  | Fail   |
| 6  | [SU][O1A] Allows creating template without config          | Fail   |
| 7  | [SU][O1A][O1O] Allows creating template with config        | Pass   |
| 8  | [SU][O1A][O1O] Allow creating VPN template                 | Pass   |
| 9  | [SU][O1A][O1O] Allow creating VPN template without openvpn | Fail   |
| 10  | [O1O] Access denied to non administrator user              | NTD    |

**SN. 4:**
![case4fail](https://user-images.githubusercontent.com/24985760/117544793-9ccce000-b040-11eb-97c5-7ceaefa516da.png)
@nemesisdesign I think this is an acceptable deviation, can you please confirm.

**SN. 5:**
![case5-1](https://user-images.githubusercontent.com/24985760/117544800-9f2f3a00-b040-11eb-9a98-4144c16494f4.png)
@ManishShah120 Can we show the error on config instead of `__all__`? it would be useful for people making apps using the API.

**SN. 9**
![case9](https://user-images.githubusercontent.com/24985760/117544802-a0606700-b040-11eb-9da2-a7b0ef22d623.png)
@nemesisdesign is this okay?

**SN. 10**
@nemesisdesign Shouldn't we deny access to the page if creating and viewing is not allowed? Same for admin UI. Why is access even allowed?

Reply:
> >    @nemesisdesign is this okay?

> It does look good to me. Why it shouldn't? I don't understand your point.

The VPN servers created from UI always have a openvpn section in configuration, here we are able to make it without that! :smile:

Reply:
@atb00ker SN5, SN6 is resolved.

Reply:
@atb00ker regarding SN10, I think you missed one more failing case which is foralong with this [O1A]  administrator but not manager i.e., (is_admin for org1 not set to True)

Reply:
Copying @nemesisdesign 's response on Gitter:

```
@ManishShah120 @atb00ker I am not sure I understand what we're discussing, can you be more explicit and provide some user stories as examples?
I am not following you.

As a general rule, for now, I'd simplify everything:

    only organization managers (having is_admin=True) or superusers can use the API
    non superusers have limited permissions, according to the permissions defined in django

So if user is not manager of the org, it can't do anything via the API right now.

If user is superuser, can do anything.

If user is not superuser and is manager of an org, it can do what their django permissions allow them to do.
```